src/celeritas/optical/Transporter.hh

```diff
@@ -0,0 +1,80 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/Transporter.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+
+#include "corecel/Types.hh"
+#include "corecel/math/NumericLimits.hh"
+#include "corecel/sys/ActionGroups.hh"
+#include "celeritas/Types.hh"
+
+#include "CoreState.hh"
+
+namespace celeritas
+{
+template<class P, template<MemSpace M> class S>
+class ActionGroups;
+
+namespace optical
+{
+class CoreParams;
+template<MemSpace M>
+class CoreState;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Transport all pending optical tracks to completion.
+ */
+class Transporter
```

Comment:
We may want to call this something else.

Reply:
Stepper maybe if we were doing a single loop, but since we're transporting all the tracks and incrementing `accum.flushes` we could call it `Flusher` 😂

Reply:
I hate it 😂 but I've got nothing better